action/config_validator.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package action
+
+import "context"
+
+// ConfigValidator describes reusable Action configuration validation functionality.
+type ConfigValidator interface {
+	// Description describes the validation in plain text formatting.
+	//
+	// This information may be automatically added to action plain text
+	// descriptions by external tooling.
+	Description(context.Context) string
+
+	// MarkdownDescription describes the validation in Markdown formatting.
+	//
+	// This information may be automatically added to action Markdown
+	// descriptions by external tooling.
+	MarkdownDescription(context.Context) string
+
+	// ValidateAction performs the validation.
+	//
+	// This method name is separate from the datasource.ConfigValidator
+	// interface ValidateDataSource method name, provider.ConfigValidator
+	// interface ValidateProvider method name, ephemeral.ConfigValidator
+	// interface ValidateEphemeralResource method name, and resource.ConfigValidator
+	// interface ValidateResource method name to allow generic validators.
```

Comment:
Super minor, probably unnecessary and may be subject to change so this does not need to be resolved now, but since we're listing all other existing `ValidateSomething` methods should this comment include `ValidateListResourceConfig` for the `list.ConfigValidator`

Reply:
Good catch, honestly not really worth keeping comments like this up-to-date, so I'll just make it more generic like the `list` package does. Fixed all the existing ones as well in f90b4f12fc0bb899fbf6e52c5868661a5f036306 ⚡